### DISCUSSION TITLE
Contribution points reference for older VSCode versions

### DISCUSSION THREAD

**Discussion by vaclavHala** [microsoft/vscode-discussions#1048] [Extension Development QnA] [answered]
Is there some way to view https://code.visualstudio.com/api/references/contribution-points for given VSCode release?
I know I can use Wayback Machine (http://web.archive.org/) to see the page at some date in history, problem with that is there is no indication in the page as to what VSCode version it talks about.
Also not all info I want can be gleaned from looking at the `@types/vscode` at desired version as that does not for instance contain info about what `contributes.menus` are supported or about what context keys are available for `when` clauses

**Comment by tamuratak** [comment, marked as answer]
You can refer to the git history of the document:

- https://github.com/microsoft/vscode-docs/blob/main/api/references/contribution-points.md

You can also search [registerExtensionPoint](https://github.com/search?q=repo%3Amicrosoft/vscode%20registerExtensionPoint&type=code) in the VS Code repository. The method registers contribution points.

**Reply by vaclavHala** [reply]
Thanks I didn't realize those pages were versioned in public git. With a bit of archaeology this should contain all the info I need. I just wish there were git tags in the repo marking releases...

**Comment by abrilurena** [comment]
Hi, @vaclavHala! The official [Contribution Points documentation](https://code.visualstudio.com/api/references/contribution-points) covers all available contribution points. The [keybindings documentation](https://code.visualstudio.com/docs/getstarted/keybindings) provides information about available keybindings and their associated contexts (when clauses). While it doesn’t explicitly mention versions, it’s a valuable resource for understanding the various extension points.

Please let me know if this helps ✨

**Reply by vaclavHala** [reply]
Hi, thanks for the response. Unfortunately this does not help me as those pages only tracks the latest vscode release. As stated in the question I want to see those pages not just for the current release but also for arbitrary release in the past
